rollout/context.go

```diff
@@ -50,7 +55,21 @@ type rolloutContext struct {
 }
 
 func (c *rolloutContext) reconcile() error {
-	err := c.checkPausedConditions()
+	// Get Rollout Validation errors
+	err := c.getRolloutValidationErrors()
+	if err != nil {
+		if vErr, ok := err.(*field.Error); ok {
+			logCtx := logutil.WithRollout(c.rollout)
+			logCtx.Info("rollout enqueue due reconcile error")
+			// We want to frequently requeue rollouts with InvalidSpec errors, because the error
+			// condition might be timing related (e.g. the Rollout was applied before the Service).
+			c.enqueueRolloutAfter(c.rollout, 20*time.Second)
+			return c.createInvalidRolloutCondition(vErr, c.rollout)
+		}
+		return err
+	}
+
+	err = c.checkPausedConditions()
```

Comment:
This logic should not exist see: https://github.com/argoproj/argo-rollouts/pull/4005/files

Reply:
and this for the original move: https://github.com/argoproj/argo-rollouts/pull/3657/files#diff-bb6b58d751a8f77a842ce48d6d247cc9541622cc9c4a5dd7d47a59e2017d6e9eL56

Reply:
I removed it. Thanks for catching that.